lib/set-npmrc-auth.js

```diff
@@ -0,0 +1,26 @@
+const {appendFile} = require('fs-extra');
+const getAuthToken = require('registry-auth-token');
+const nerfDart = require('nerf-dart');
+const SemanticReleaseError = require('@semantic-release/error');
+const getRegistry = require('./get-registry');
+
+module.exports = async ({publishConfig, name}, logger) => {
+  const registry = await getRegistry(publishConfig, name);
+  logger.log('Verify authentication for registry %s', registry);
+  const {NPM_TOKEN, NPM_USERNAME, NPM_PASSWORD, NPM_EMAIL} = process.env;
+
+  if (!getAuthToken(registry)) {
```

Comment:
maybe return early instead of wrapping the rest of the code into the if statemen?

```js
if (getAuthToken(registry)) {
  return
}
```

Reply:
Done